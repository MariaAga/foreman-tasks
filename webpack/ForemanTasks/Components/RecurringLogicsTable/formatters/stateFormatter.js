import React from 'react';
import { cellFormatter } from 'foremanReact/components/common/table';
import { translate as __ } from 'foremanReact/common/I18n';

const formatText = state => {
  switch (state) {
    case 'active':
      return __('Active');
    case 'cancelled':
      return __('Cancelled');
    case 'finished':
      return __('Finished');
    case 'disabled':
      return __('Disabled');
    default:
      return __('N/A');
  }
};

export const stateFormatter = state => {
  const icons = {
    active: 'glyphicon-info-sign',
    disabled: 'glyphicon-pause',
    finished: 'glyphicon-ok-sign',
    cancelled: 'glyphicon-warning-sign',
  };
  const status = {
    finished: 'status-ok',
    cancelled: 'status-error',
  };
  const defaultIcon = 'glyphicon-question-sign';
  return cellFormatter(
    <React.Fragment>
      <i className={`glyphicon ${icons[state] || defaultIcon}`} />
      <span className={status[state]}>{formatText(state)}</span>
    </React.Fragment>
  );
};
