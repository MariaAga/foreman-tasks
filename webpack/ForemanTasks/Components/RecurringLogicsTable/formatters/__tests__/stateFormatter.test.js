import { stateFormatter } from '../stateFormatter';

describe('stateFormatter', () => {
  it('render disabled', () => {
    expect(stateFormatter('disabled')).toMatchSnapshot();
  });
  it('render active', () => {
    expect(stateFormatter('active')).toMatchSnapshot();
  });
  it('render finished', () => {
    expect(stateFormatter('finished')).toMatchSnapshot();
  });
  it('render cancelled', () => {
    expect(stateFormatter('cancelled')).toMatchSnapshot();
  });
});
