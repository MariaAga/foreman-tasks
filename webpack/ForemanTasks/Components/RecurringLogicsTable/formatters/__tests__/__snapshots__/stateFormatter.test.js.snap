// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stateFormatter render active 1`] = `
<React.Fragment>
  <i
    className="glyphicon glyphicon-info-sign"
  />
  <span>
    Active
  </span>
</React.Fragment>
`;

exports[`stateFormatter render cancelled 1`] = `
<React.Fragment>
  <i
    className="glyphicon glyphicon-warning-sign"
  />
  <span
    className="status-error"
  >
    Cancelled
  </span>
</React.Fragment>
`;

exports[`stateFormatter render disabled 1`] = `
<React.Fragment>
  <i
    className="glyphicon glyphicon-pause"
  />
  <span>
    Disabled
  </span>
</React.Fragment>
`;

exports[`stateFormatter render finished 1`] = `
<React.Fragment>
  <i
    className="glyphicon glyphicon-ok-sign"
  />
  <span
    className="status-ok"
  >
    Finished
  </span>
</React.Fragment>
`;
